### DISCUSSION TITLE
Download all documents together without checking collection

### DISCUSSION THREAD

**Discussion by AnoopJoze** [spatie/laravel-medialibrary#2984] [Q&A] [answered]
I'm trying to download all the documents together that belong to one user. If one user has uploaded documents under multiple collections, I want to download all uploads under all collections in one click as a zip file.

I tried without the collection name in getMedia function. But it's not getting my result.

$user = User::where('id',auth()->user()->id)->first();
$downloads = $user->getMedia();
return MediaStream::create('my-files.zip')->addMedia($downloads);

**Comment by erikn69** [comment]
Use the [Media](https://github.com/spatie/laravel-medialibrary/blob/main/src/MediaCollections/Models/Media.php) model and filter the user on the relation, for example using `whereHasMorph`

**Reply by AnoopJoze** [reply]
@erikn69  Like this? 
$user = User::where('id',auth()->user()->id)->first();
$downloads = Media::where('model_id',$user->id)->get();
return MediaStream::create('myfiles.zip')->addMedia($downloads);

Im getting error for this.

**Reply by AnoopJoze** [reply]
I created a separate model for Media, that was the error.

**Reply by erikn69** [reply]
Seriously, you put yourself as an answer, ok, this is the last time I help people, they always do the same.

**Reply by parallels999** [reply]
There are many disrespectful people, then they complain because in the open source community no one helps anyone

**Comment by AnoopJoze** [comment, marked as answer]
I got it. thank you @erikn69 

$user = User::where('id',auth()->user()->id)->first();
$downloads = ModelsMedia::where('model_id',$user->id)->get();
return MediaStream::create('myfiles.zip')->addMedia($downloads);

**Reply by erikn69** [reply]
Also this answer has fails
